sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -32,9 +51,26 @@ public final class SentryPerformanceProvider extends EmptySecureContentProvider
   private @Nullable Application app;
   private @Nullable Application.ActivityLifecycleCallbacks activityCallback;
 
+  private final @NotNull ILogger logger;
+  private final @NotNull BuildInfoProvider buildInfoProvider;
+
+  @TestOnly
+  SentryPerformanceProvider(
+      final @NotNull ILogger logger, final @NotNull BuildInfoProvider buildInfoProvider) {
+    this.logger = logger;
+    this.buildInfoProvider = buildInfoProvider;
+  }
+
+  public SentryPerformanceProvider() {
+    logger = new AndroidLogger();
+    buildInfoProvider = new BuildInfoProvider(logger);
+  }
+
   @Override
   public boolean onCreate() {
-    onAppLaunched(getContext());
+    final @NotNull AppStartMetrics appStartMetrics = AppStartMetrics.getInstance();
+    launchStartupProfiler(appStartMetrics);
+    onAppLaunched(getContext(), appStartMetrics);
```

Comment:
`h`: I think we should probably swap this with `launchStartupProfiler`, to make it how it was before merging the two content providers. `onAppLaunched` reports a high-level metric which will probably be surfaced to the user somewhere in the beginning of the Starfish workflow, and if we don't include what it takes to launch the profiler we're gonna be lying (especially because `launchStartupProfiler` does file i/o and json deser) about the app startup metric.

You probably wanted to include `onAppLaunched` into the profile as well, but there really shouldn't be anything that takes a lot of time.

Reply:
> You probably wanted to include onAppLaunched into the profile as well, but there really shouldn't be anything that takes a lot of time.
Yeah, that was the reason, and it was it was working with 2 different providers. But it's fine to swap them, as the profiler launch should be the slowest of the two